tools/sdk/resourcemanager/models.go

```diff
@@ -5,9 +5,10 @@ import (
 )
 
 type ApiObjectDefinition struct {
-	NestedItem    *ApiObjectDefinition    `json:"nestedItem,omitempty"`
-	ReferenceName *string                 `json:"referenceName,omitempty"`
-	Type          ApiObjectDefinitionType `json:"type"`
+	NestedItem            *ApiObjectDefinition    `json:"nestedItem,omitempty"`
+	ReferenceName         *string                 `json:"referenceName,omitempty"`
+	ReferenceIsCommonType *bool                   `json:"referenceIsCommonType,omitempty"`
```

Comment:
(as above) do we need this at this level? if we instead pulled the settings from the API we'd achieve the same thing, which might actually make both generators easier? e.g.

```
GET /services
{
  "microsoft-graph": {
    "rootUri": "/v1/microsoft-graph",
    "usesCommonTypes": true
  }
  "resource-manager": {
    "rootUri": "/v1/resource-manager",
    "usesCommonTypes": false
  }
}
```

Which then means the CLI itself could take an argument expecting the Data Source (e.g. Microsoft Graph/Resource Manager) pulled from the API? This has the benefit of features being in one place (but a short-time compile-time hit until we're on v2 of the Data API)